clusterloader2/pkg/config/codec.go

```diff
@@ -21,13 +21,19 @@ import (
 	"io/ioutil"
 	"path/filepath"
 	"strings"
+	"sync"
 
 	"github.com/spf13/viper"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/perf-tests/clusterloader2/api"
 )
 
+var (
+	lock          sync.RWMutex
```

Comment:
Instead of defining global variables that are shared across the package, could you please move this into a separate TemplateCache struct? That way it'll also protect the purpose of the lock.

Reply:
By "move this into a separate TemplateCache struct" do you mean creating separate struct to have it encapsulated and still have singleton of this struct on global level?
Or do you want this struct to have something like create function so we can create different instances of cache struct?

Reply:
The latter.

Reply:
OK. So the consequence of this will be having cache stored in the test context - cache needs to be passed between different calls.

Reply:
Sure, that's better than having one globally usable instance of the package. Alternatively, if you want to avoid passing it across calls, you can instantiate a global variable of this struct type in that package.

Reply:
What about `TemplateProvider`? Because it's not only cache but also something that can read template from file system.

Reply:
I don't have any strong preference here :)

Reply:
OK. Then `TemplateProvider` it is.